### PR TITLE
Bugfix/Deleting a participation with submissions + results

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] I documented my source code using the JavaDoc / JSDoc style.
- [x] I added integration test cases for the server (Spring) related to the features
- [x] ~I added integration test cases for the client (Jest) related to the features~
- [x] ~I added screenshots/screencast of my UI changes~
- [x] ~I translated all the newly inserted strings~

### Description

Regression from https://github.com/ls1intum/Artemis/pull/695.
If a programming exercise participation is deleted that has a submission with a result, the deletion fails with this error:
```
Caused by: java.sql.SQLIntegrityConstraintViolationException: Cannot delete or update a parent row: a foreign key constraint fails (`ArTEMiS`.`result`, CONSTRAINT `FK3vct9sad5oubthdmq63n58mnp` FOREIGN KEY (`submission_id`) REFERENCES `submission` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION)
```

So the issue is that Hibernate tries to delete the submission before the result, which causes the foreign key constraint.

I made 3 small changes:
- Eagerly load the results and submissions in the delete service method (otherwise they appearently can't be loaded correctly in `deleteResultsAndSubmissionsOfParticipation`.
- Moved the logic for deleting submissions without a result into `deleteResultsAndSubmissionsOfParticipation`.
- Return the cleaned up participation without the submissions and results; otherwise Hibernate seems to try to delete the connected entities again when the participation is deleted. I think the issue is that there a 2 transactional contexts on top of each other here (1: delete, 2: deleteResultsAndSubmissionsOfParticipation)

I also wrote some integration tests & updated the docs.

:information_source: It is unclear to me why this isn't an issue for e.g. quizzes.

### Steps for Testing

1. Log in to ArTEMiS
2. Course administration > programming exercise > edit in editor
3. Create assignment participation
4. Submit for the assignment participation, wait for the result to return
5. Reset assignment repository (this first deletes the participation and then creates a new one)
-> There should be no error shown

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java [hunk 1]
```diff
@@ -600,7 +600,7 @@ else if (participation.getExercise() instanceof ProgrammingExercise) {
      */
     @PutMapping("/participations/{id}/cleanupBuildPlan")
     @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Participation> deleteParticipation(@PathVariable Long id, Principal principal) {
+    public ResponseEntity<Participation> cleanupBuildPlan(@PathVariable Long id, Principal principal) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]:
```suggestion
    public ResponseEntity<Participation> cleanupBuildPlan(@PathVariable Long participationId, Principal principal) {
```

File: src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java [hunk 2]
```diff
@@ -600,7 +600,7 @@ else if (participation.getExercise() instanceof ProgrammingExercise) {
...
         ProgrammingExerciseStudentParticipation participation = (ProgrammingExerciseStudentParticipation) participationService.findOneStudentParticipation(id);
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java]:
```suggestion
        ProgrammingExerciseStudentParticipation participation = (ProgrammingExerciseStudentParticipation) participationService.findOneStudentParticipation(participationId);
```

File: src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java [hunk 3]
```diff
@@ -846,11 +841,18 @@ public void deleteResultsAndSubmissionsOfParticipation(Participation participati
                     Submission submissionToDelete = result.getSubmission();
                     submissionRepository.deleteById(submissionToDelete.getId());
                     result.setSubmission(null);
-                    // make sure submissions don't get deleted twice (see below)
                     participation.removeSubmissions(submissionToDelete);
                 }
             }
         }
+        // The following case is necessary, because we might have submissions without a result. At this point only submissions without a result will still be connected to the
```
Comment (lovaiible) [inline src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java]: @thilo-behnke Looks good to me! Just one minor remark: You have covered participations without a result here, so maybe it's good to have this case covered in the unit tests as well?

Comment (thilo-behnke) [reply src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java]: You are right, I updated the test.

**Pull request merged**